crates/paimon/src/spec/schema_change.rs

```diff
@@ -0,0 +1,519 @@
+    ///
+    /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L95>
+    SetOption { key: String, value: String },
+    /// A SchemaChange to remove a table option.
+    ///
+    /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L134>
+    RemoveOption { key: String },
+    /// A SchemaChange to update a table comment.
+    ///
+    /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L167>
+    UpdateComment { comment: Option<String> },
+    /// A SchemaChange to add a new field.
+    ///
+    /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L201>
+    AddColumn {
+        field_name: String,
+        data_type: DataType,
+        description: Option<String>,
+        #[serde(rename = "move")]
+        column_move: Option<ColumnMove>,
+    },
+    /// A SchemaChange to rename a field.
+    ///
+    /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L260>
+    RenameColumn {
+        field_name: String,
+        new_name: String,
+    },
+    /// A SchemaChange to drop a field.
+    ///
+    /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L302>
+    DropColumn { field_name: String },
+    /// A SchemaChange to update the field's type.
+    ///
+    /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L335>
+    UpdateColumnType {
+        field_name: String,
+        data_type: DataType,
+    },
+    /// A SchemaChange to update the field's position.
+    ///
+    /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L377>
+    UpdateColumnPosition {
+        #[serde(rename = "move")]
+        column_move: ColumnMove,
+    },
+    /// A SchemaChange to update the field's nullability.
+    ///
+    /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L470>
+    UpdateColumnNullability {
+        field_name: Vec<String>,
+        nullable: bool,
+    },
+    /// A SchemaChange to update the (nested) field's comment.
+    ///
+    /// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L512>
+    UpdateColumnComment {
+        field_names: Vec<String>,
+        new_description: String,
+    },
+}
+
+impl SchemaChange {
+    /// impl the `set_option`.
+    pub fn set_option(key: String, value: String) -> Self {
```

Comment:
I feel like we can make all fields in this enum to be `pub` so that users can construct via `SchemaChange::SetOption {key, value}`.

Reply:
When I change fields in the enum  to `pub`, which cause error like this: `note: enum variants and their fields always share the visibility of the enum they are in`. Do you have some suggestions to make all fields in this enum to be `pub`?

Reply:
> Do you have some suggestions to make all fields in this enum to be `pub`?

Yep, I think it's fine to make those fields public.